src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -665,6 +665,9 @@ void Creature::Update(uint32 diff)
 {
     if (IsAIEnabled() && m_triggerJustAppeared && m_deathState != DEAD)
     {
+        if (IsAreaSpiritHealer())
+            CastSpell(nullptr, SPELL_SPIRIT_HEAL_CHANNEL_AOE, false);
```

Comment:
You're moving everything to scripts and spell system, yet this is here, this belongs to a script